silkrpc/commands/eth_api.cpp

```diff
@@ -296,12 +296,40 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_uncle_by_block_hash_and_ind
 
 // https://eth.wiki/json-rpc/API#eth_getunclebyblocknumberandindex
 asio::awaitable<void> EthereumRpcApi::handle_eth_get_uncle_by_block_number_and_index(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    if (params.size() != 2) {
+        auto error_msg = "invalid eth_getUncleByBlockNumberAndIndex params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+    const auto block_id = params[0].get<std::string>();
+    const auto index = params[1].get<std::string>();
+    SILKRPC_DEBUG << "block_id: " << block_id << " index: " << index << "\n";
+
     auto tx = co_await database_->begin();
 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"],  "0x" + to_hex_no_leading_zeros(0));
+        const auto block_number = co_await core::get_block_number(block_id, tx_database);
+        const auto block_with_hash = co_await core::rawdb::read_block_by_number(tx_database, block_number);
+        const auto ommers = block_with_hash.block.ommers;
+
+        auto idx = std::stoul(index);
```

Comment:
Have you tried with `index` different from `0x0` (e.g. `0x1`)? This line doesn't work, it should be:

`auto idx = std::stoul(index, 0, 16);`

because `index` is in QUANTITY format (we should introduce a new `silkrpc::Quantity` type but we'll do in a separate PR)